internal/keysharecore/usersecrets.go

```diff
@@ -141,19 +182,15 @@ func (c *Core) decryptUserSecrets(secrets UserSecrets) (unencryptedUserSecrets,
 }
 
 func (c *Core) decryptUserSecretsIfPinOK(secrets UserSecrets, pin string) (unencryptedUserSecrets, error) {
-	paddedPin, err := padBytes([]byte(pin), 64)
+	s, err := c.decryptUserSecrets(secrets)
```

Comment:
I'd rather keep this, then. Having this method is I think a little safer, because it makes it more likely that we don't accidentally do something with the user secrets while we shouldn't because the PIN doesn't match. Additionally, your suggestion would add a little code duplication.